internal/impl/devbox.go

```diff
@@ -598,7 +607,48 @@ func (d *Devbox) ensurePackagesAreInstalled(mode installMode) error {
 	}
 	fmt.Fprintln(d.writer, "Done.")
 
-	return plugin.RemoveInvalidSymlinks(d.projectDir)
+	err := plugin.RemoveInvalidSymlinks(d.projectDir)
+	if err != nil {
+		return err
+	}
+
+	if err := os.WriteFile(hashFile, hash, 0644); err != nil {
+		debug.Log("error writing hash file: %s", err)
```

Comment:
if we cannot update the hash (i.e. `err != nil`), then the hashfile is invalid and old so should we try to `os.Remove` it inside this if-block?